From 077fbdda1c70413b6ff1af26f5ee0a7b814aa934 Mon Sep 17 00:00:00 2001
From: Paul Walker <paul@pwjw.com>
Date: Sun, 25 Apr 2021 09:39:20 -0400
Subject: [PATCH] The DSP Test Fix from XT into the main branch also

Basically: Make sure the test is specific about its
file
---
 src/headless/UnitTestsDSP.cpp | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/src/headless/UnitTestsDSP.cpp b/src/headless/UnitTestsDSP.cpp
index b87ceb4f8b4..72f2fb83239 100644
--- a/src/headless/UnitTestsDSP.cpp
+++ b/src/headless/UnitTestsDSP.cpp
@@ -792,6 +792,25 @@ TEST_CASE("Every Oscillator Plays", "[dsp]")
 
             for (int q = 0; q < 10; ++q)
                 surge->process();
+
+            int idx = 0;
+            bool got = false;
+            for (auto q : surge->storage.wt_list)
+            {
+                if (q.name == "Sine Power HQ")
+                {
+                    got = true;
+                    surge->storage.getPatch().scene[0].osc[0].wt.queue_id = idx;
+                }
+                idx++;
+            }
+            REQUIRE(got);
+            for (int q = 0; q < 10; ++q)
+                surge->process();
+
+            REQUIRE(std::string(surge->storage.getPatch().scene[0].osc[0].wavetable_display_name) ==
+                    "Sine Power HQ");
+
             float sumAbsOut = 0;
             surge->playNote(0, 60, 127, 0);
             for (int q = 0; q < 100; ++q)
